distributed/tests/test_worker.py

```diff
@@ -1180,33 +1180,39 @@ async def test_statistical_profiling_2(c, s, a, b):
 
 
 @gen_cluster(
-    nthreads=[("127.0.0.1", 1)],
+    nthreads=[("", 1)],
     client=True,
-    worker_kwargs={"memory_monitor_interval": 10},
+    config={
+        "distributed.worker.memory.target": False,
+        "distributed.worker.memory.spill": 0.7,
+    },
+    worker_kwargs={"memory_monitor_interval": "10ms"},
 )
 async def test_robust_to_bad_sizeof_estimates(c, s, a):
-    np = pytest.importorskip("numpy")
-    memory = psutil.Process().memory_info().rss
+    """Test that the spill threshold uses the process memory and not the managed memory
+    reported by sizeof(), which may be inaccurate
+    """
+    memory = s.workers[a.address].memory.process
+    # Reach 'spill' threshold after 400MB of managed data
     a.memory_limit = memory / 0.7 + 400e6
```

Comment:
Note that this works just by luck today for the purpose of the spill threshold, and does NOT work for the target threshold. See #5367